File: app/src/main/java/com/elvotra/clean/threading/DiskIOThreadExecutor.java
package com.elvotra.clean.threading;

import android.support.annotation.NonNull;

import java.util.concurrent.Executor;
import java.util.concurrent.Executors;

public class DiskIOThreadExecutor implements Executor {

    private final Executor executor;

    public DiskIOThreadExecutor() {
        executor = Executors.newSingleThreadExecutor();
    }

    @Override
    public void execute(@NonNull Runnable command) {
        executor.execute(command);
    }
}

File: app/src/main/java/com/elvotra/clean/threading/DiskIOThreadExecutor.kt
package com.elvotra.clean.threading

import java.util.concurrent.Executor
import java.util.concurrent.Executors

class DiskIOThreadExecutor : Executor {

    private val executor: Executor

    init {
        executor = Executors.newSingleThreadExecutor()
    }

    override fun execute(command: Runnable) {
        executor.execute(command)
    }
}